app/contracts/users/update_contract.rb

```diff
@@ -49,7 +49,11 @@ def user_allowed_to_update
     end
 
     def at_least_one_admin_is_active
-      if (model.locked? || !model.admin?) && User.active.admin.where.not(id: model).none?
+      return unless
+        (model.admin_changed? && !model.admin?) ||
+        (model.status_changed? && model.locked?)
```

Comment:
Wouldn't this need to be 

```ruby
(model.admin_changed? && !model.admin?) ||
        (model.admin && model.status_changed? && model.locked?)
```

because you only want to catch admins being locked?

Reply:
Nice catch @oliverguenther, thanks!